core/src/test/java/com/backblaze/b2/json/B2JsonBoundedByteArrayOutputStreamTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+package com.backblaze.b2.json;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+
+import static junit.framework.Assert.assertNull;
+import static junit.framework.TestCase.assertEquals;
+
+/**
+ * Unit test for B2JsonByteArrayOutputStream
+ */
+public class B2JsonBoundedByteArrayOutputStreamTest {
```

Comment:
Does this test have full coverage of the code?

In addition to full coverage, it would be good to test:

- empty results
- writing empty arrays
- boundary conditions at expansion size
- multiple expansions
- a result that is exactly the maximum size, written as one block
- a result that is exactly the maximum size, written one byte at a time

Reply:
Added full coverage on public methods, and additional tests you mentioned (pretty much I think).

Reply:
Is that full coverage of the entire class?  All the lines in the source file?